Plugins/Public/cloak_plugin/Cloak.cpp

```diff
@@ -291,51 +309,40 @@ void SetState(uint iClientID, uint iShipID, int iNewState)
 	}
 }
 
-bool removeSingleFuel(uint iClientID, CLOAK_INFO &info)
-{
-	for (list<EquipDesc>::iterator item = Players[iClientID].equipDescList.equip.begin(); item != Players[iClientID].equipDescList.equip.end(); item++)
-	{
-		uint fuel_usage = info.arch.mapFuelToUsage[item->iArchID];
-		if (item->iCount >= fuel_usage)
-		{
-			pub::Player::RemoveCargo(iClientID, item->sID, fuel_usage);
-			info.singleCloakConsumed = true;
-			return true;
-		}
-	}
-
-	return false;
-
-}
-
 // Returns false if the ship has no fuel to operate its cloaking device.
-static bool ProcessFuel(uint iClientID, CLOAK_INFO &info)
+static bool ProcessFuel(uint iClientID, CLOAK_INFO &info, uint iShipID)
 {
 	if (info.bAdmin)
 		return true;
 
-	if (!info.singleCloakConsumed && info.arch.singleUseCloak) {
-		return removeSingleFuel(iClientID, info);
-	}
-
-	else if (info.singleCloakConsumed)
-	{
+	if(setJumpingClients.find(iClientID) != setJumpingClients.end())
 		return true;
-	}
 
 	for (list<EquipDesc>::iterator item = Players[iClientID].equipDescList.equip.begin(); item != Players[iClientID].equipDescList.equip.end(); item++)
 	{
 		if (info.arch.mapFuelToUsage.find(item->iArchID) != info.arch.mapFuelToUsage.end())
 		{
-			uint fuel_usage = info.arch.mapFuelToUsage[item->iArchID];
-			if (item->iCount >= fuel_usage)
+			const auto& fuelUsage = info.arch.mapFuelToUsage[item->iArchID];
+			float currFuelUsage = fuelUsage.usageStatic;
+			if (fuelUsage.usageLinear != 0.0f || fuelUsage.usageSquare != 0.0f) {
+				Vector dir1;
+				Vector dir2;
+				pub::SpaceObj::GetMotion(iShipID, dir1, dir2);
+				float vecLength = sqrtf(dir1.x * dir1.x + dir1.y * dir1.y + dir1.z * dir1.z);
+				
+				currFuelUsage += fuelUsage.usageLinear * vecLength;
+				currFuelUsage += fuelUsage.usageSquare * vecLength * vecLength;

```

Comment:
so wait... we want to add fuel usage based on speed... but also based on speed squared?

Reply:
Correct, we want the formula to scale exponentially with speed, requiring us to use a quadratic formula to achieve wanted consumption at specific speed breakpoints. It also enables the player to traverse a larger distance if they're willing to move at lower speed.

Reply:
very well